### PR TITLE
AzureMonitor: resourcePicker should show resource in the correct subs…

### LINKED ISSUE #40889: Resource picker doesn't show correct data when identity has access to resources in multiple subscriptions
**What happened**:
Using Azure Monitor data source.
Configured managed identity with Monitor Reader role to one subscription(1), and to a Log Analytics workspace in a second subscription (2).
When building a query with Data Source: Azure Monitor and Service: Logs, the Select a **resource scope incorrectly shows the resources for subscription (1) under subscription (2)** 

**What you expected to happen**:
Resources for a subscription should only be listed under the subscription to which they belong.
image

**How to reproduce it (as minimally and precisely as possible)**:
Configure Azure Monitor Data source
Allow Monitor reader access to one subscription and to a single resource in another subscription.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.5
- Data source type & version:  Azure Monitor 0.30
- OS Grafana is installed on:
- User OS & Browser: Microsoft Edge
- Grafana plugins: Azure Monitor
- Others:

### PR DESCRIPTION
**Pull request by iwysiu**
…cription

**What this PR does / why we need it**:
Fixes an error where if two subscriptions have a resource group with the same name, it would show up twice under both.

**Which issue(s) this PR fixes**:

Fixes #40889 

**Special notes for your reviewer**:
The test doesn't actually replicate the bug (it had to do with the azure resource graph query) but I thought it would be good to have.

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.test.ts [hunk 1]
```diff
@@ -30,21 +30,22 @@ describe('AzureMonitor resourcePickerData', () => {
     it('returns only subscriptions at the top level', async () => {
       const results = await resourcePickerData.getResourcePickerData();
 
-      expect(results.map((v) => v.id)).toEqual(['/subscriptions/abc-123', '/subscription/def-456']);
+      expect(results.map((v) => v.id)).toEqual(['/subscriptions/abc-123', '/subscriptions/def-456']);
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.test.ts:33]: Maybe you can add in the test above ( line 27) a check that verifies that the query contains `| project resourceGroupURI=id, resourceGroupName=name, resourceGroup, subscriptionId`? Just to have a test that covers the new code

**Review by andresmgot — APPROVED** [review]
Great job replicating the issue!

**Comment by VRay27** [comment]
Has this been fixed in the 8.3.0 (2021-11-30) release?

**Comment by andresmgot** [comment]
Thanks for raising this @VRay27, we forgot to add the backport labels so this didn't make it to 8.3.0. Let me add those now so it can be included in 8.3.1.

**Comment by VRay27** [comment]
@andresmgot  Could you please confirm the date when this 8.3.1 will be released?

**Comment by andresmgot** [comment]
@VRay27 we don't have a fixed date for that yet. The usual release cadence for patch versions is one week (it could happen sooner than that).

**Comment by VRay27** [comment]
@andresmgot  I can see the 8.3.1 has been released but I don't see the #42170 in it's changelog. Could you please confirm when this will released?
image

**Comment by andresmgot** [comment]
@VRay27 I can confirm that this has been released in 8.3.1.

**Comment by andresmgot** [comment]
@VRay27 actually no, since it was a security release, only the security patch was included in the new version. The fix will be available in the next release (8.3.2), that will be released soon (in the following days)

**Comment by VRay27** [comment]
@andresmgot @iwysiu  Again 8.3.3. has been released but this bug is not part of that. I not seeing this bug included in the release changelog
https://github.com/grafana/grafana/blob/main/CHANGELOG.md

**Comment by andresmgot** [comment]
@VRay27 it's not in the changelog but the fix is there: https://github.com/grafana/grafana/blob/v8.3.3/public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts#L49
Let us know if that doesn't work for you.

**Pull request merged**